.github/actions/build-pgo-wheel/action.yml

```diff
@@ -0,0 +1,74 @@
+name: Build PGO wheel
+description: Builds a PGO-optimized wheel
+inputs:
+  interpreter:
+    description: 'Interpreter to build the wheel for'
+    required: true
+  rust-toolchain:
+    description: 'Rust toolchain to use'
+    required: true
+outputs:
+  wheel:
+    description: 'Path to the built wheel'
+    value: ${{ steps.find_wheel.outputs.path }}
+runs:
+  using: "composite"
+  steps:
+    - name: prepare self schema
+      shell: bash
+      # generate up front so that we don't have to do this inside the docker container
+      run: uv run python generate_self_schema.py
+
+    - name: prepare profiling directory
+      shell: bash
+      # making this ahead of the compile ensures that the local user can write to this
+      # directory; the maturin action (on linux) runs in docker so would create as root
+      run: mkdir -p ${{ github.workspace }}/profdata
```

Comment:
This turned out to be the critical change; as per the comment without this the linux jobs (which are built inside `manylinux` docker containers) were failing to write out their profiling information.

Reply:
is there a way to build with "require profiling information" so build fails if it can't be found?

Reply:
The annoying thing is that the build already emits profiling information for the proc macros, so there will always be some. I'm also not sure of a way to enforce. 

At least with the way CI is now set up, if the PGO breaks we will see codspeed regressions.